tests/integration_tests/test_dataset.py

```diff
@@ -25,6 +25,47 @@ def test_construct_text_corpus():
     dataset.construct_text_corpus()
     dataset.close()
 
+def test_change_data_size():
```

Comment:
This is a good test. One thing I'd suggest is to also test your failure cases. That is, I see you have a local function called `check_valid` in there. You should try to trigger that as well. IMO you should just raise a `ValueError`, and use an `assertRaises`. Check https://stackoverflow.com/a/61017213

Reply:
I dont really know what this mean, perhaps clarify so i can get a better understanding. I did add two tests that check the inputs that are tokenized and not tokenized under the generator and not the generator.